source/Dialog.h

```diff
@@ -44,13 +44,12 @@ class Dialog : public Panel {
 	// Three different kinds of dialogs can be constructed: requesting numerical
 	// input, requesting text input, or not requesting any input at all. In any
 	// case, the callback is called only if the user selects "ok", not "cancel."
-template <class T>
-	Dialog(T *t, void (T::*fun)(int), const std::string &text);
 template <class T>
 	Dialog(T *t, void (T::*fun)(int), const std::string &text, int initialValue);
 	
 template <class T>
-	Dialog(T *t, void (T::*fun)(const std::string &), const std::string &text);
+	Dialog(T *t, void (T::*fun)(const std::string &), const std::string &text,
+			const std::string& initialValue = "");
```

Comment:
Lose the `const &` from the initial value here, so that the compiler can omit copying in some cases. This can also be on the same line as above.

Also, is a default value necessary (are there dialogs with a callback that takes a string but don't provide an initial value)?

Reply:
> Lose the `const &` from the initial value here, so that the compiler can omit copying in some cases. This can also be on the same line as above.

Done.

> Also, is a default value necessary (are there dialogs with a callback that takes a string but don't provide an initial value)?

Yes, there are such dialogs:

https://github.com/endless-sky/endless-sky/blob/82487024212a7a1f93fad982048a933fe6d83631/source/BankPanel.cpp#L269-L275

Should the default value be kept then? Or do you see another approach?